aten/src/ATen/record_function.cpp

```diff
@@ -525,7 +606,7 @@ void bumpRecordAllFunctions() {
 }
 
 void releaseRecordAllFunctions() {
-  TORCH_CHECK(global_record_all_functions_.fetch_sub(1, std::memory_order_relaxed) >= 0);
+  TORCH_CHECK(global_record_all_functions_.fetch_sub(1, std::memory_order_relaxed) > 0);
```

Comment:
why do we disallow zero here?

Reply:
because fetch_sub returns the old value. if the old value was 0, the new value is -1, which is incorrect.